test/16_recode.jl

```diff
@@ -534,4 +538,70 @@ end
     @test !isordered(y)
 end
 
+@testset "replace with CategoricalArray" begin
+    function testf(f, T, x::CategoricalArray, y, pairs::Pair...)
+        ca = f(x, pairs...)
+        @test ca ≅ y
+        @test ca isa CategoricalArray{T}
+
+        if VERSION >= v"0.7.0-"
+            a = f(Array(x), pairs...)
+            @test a ≅ y
+            @test a isa Array{T}
+        end
+
+        ca
+    end
+
+    @testset "string with missings" begin
+        x = categorical(["a", "b", missing, "a"])
```

Comment:
Can you also test a situation where the input doesn't allow for `missing`, but the replace operation introduces one?